webapp/app/js/compare.js

```diff
@@ -0,0 +1,448 @@
+
+          // find summary results for all tests/platforms for the original rev
+          var signatureURL = thServiceDomain + '/api/project/' +
+              $scope.originalProject + '/performance-data/0/' +
+              'get_performance_data/?interval_seconds=' + $scope.timeRange;
+
+          // TODO: figure how how to reduce these maps
+          var rawResultsMap = {};
+
+          $q.all(seriesList.map(function(series) {
+            return $http.get(signatureURL + "&signatures=" + series.signature).then(function(response) {
+              response.data.forEach(function(data) {
+                rawResultsMap[data.series_signature] = calculateStats(data.blob, $scope.originalResultSetID);
+              });
+            });
+          })).then(function () {
+
+            // find summary results for all tests/platforms for the original rev
+            var signatureURL = thServiceDomain + '/api/project/' +
+                           $scope.newProject + '/performance-data/0/' +
+                           'get_performance_data/?interval_seconds=' + $scope.timeRange;
+
+            //ok, now get the new revision
+            var signatureListURL = thServiceDomain + '/api/project/' + $scope.newProject + 
+              '/performance-data/0/get_performance_series_summary/?interval=' +
+              $scope.timeRange;
+
+            // TODO: figure how how to reduce these maps
+            var new_rawResultsMap = {};
+            var new_seriesList = [];
+
+
+            $http.get(signatureListURL).then(function(response) {
+              Object.keys(response.data).forEach(function(signature) {
+                var seriesSummary = getSeriesSummary(signature,
+                                                     response.data[signature],
+                                                     optionCollectionMap,
+                                                     $stateParams.pgo,
+                                                     $stateParams.e10s);
+
+                if (seriesSummary != null && seriesSummary.signature !== undefined) {
+                  new_seriesList.push(seriesSummary);
+                }
+              });
+
+
+              $q.all(new_seriesList.map(function(series) {
+                return $http.get(signatureURL + "&signatures=" + series.signature).then(function(response) {
+                  response.data.forEach(function(data) {
+                    new_rawResultsMap[data.series_signature] = calculateStats(data.blob, $scope.newResultSetID);
+                  });
+                });
+              })).then(function () {
+                displayResults(seriesList, rawResultsMap, new_seriesList, new_rawResultsMap);
+              });
+            });
+          });
+        }
+      );
+    }
+
+    function calculateStats(perfData, resultSetID) {
+      var geomeans = [];
+      var total = 0;
+      perfData.forEach(function(pdata) {
+        if (pdata.result_set_id != resultSetID) {
```

Comment:
Change this to if (pdata.result_set_id === resultSetID) { , then push the geomean/total updating stuff in here. That way we can avoid returning null.